docs/Behaviour - Server Side.md

```diff
@@ -23,26 +25,30 @@ When a Sender is in this state, ie. responding with a `205 Reset Content`, this
 ### Request Methods
 
 - A `GET` request returns the last successfully applied Media Profiles array.
-- A `PUT` request MUST be validated and if the Device supports the proposed media profile, it initiates an attempt to reconfigure the Device and update corresponding Flow, Source, and/or Sender properties. If successful, the operation MUST return the accepted Media Profiles array, otherwise an error MUST be returned and nothing changed.
+- A `PUT` request MUST be validated and if the Sender supports the proposed Media Profile, the Inputs are started to be reconfigured which causes updating of the corresponding Flow, Source, and/or Sender properties. If successful, the operation MUST return the accepted Media Profiles array, otherwise an error MUST be returned and nothing changed.
 - A `DELETE` request MUST clear the last successfully applied Media Profiles array. It turns Media Profiles to the initial state which exclude Sink Metadata Processing out of the process. It is RECOMMENDED to be performed by a client after disconnecting all Receivers from an according Sender.
 
 When the `/media-profiles` structure changes as a consequence of PUT or DELETE (request is validated and accepted by the API server), then the associated [IS-04][IS-04] Sender MUST have its version updated (in registered mode this MUST update the registered resource).
 
 ### Integration with IS-04
 
-If the new `/media-profiles` structure causes the Sender or underlying [IS-04][IS-04] Flow and Source to change, then their version MUST be updated (in registered mode this MUST update the registered resources).
+If the new `/media-profiles` structure causes the Sender or underlying Flow, Source or Inputs to change, then their versions MUST be updated (in registered mode this MUST update the registered resources).
 
 ## Receivers
 
-The `/sinks` endpoint returns a list of IDs corresponding to Sinks associated to the specific Receiver. If the list of associated sinks changes then the underlying [IS-04][IS-04] Receiver MUST update its version (in registered mode this MUST update the registered resource).
+The `/outputs` endpoint returns a list of IDs corresponding to Outputs associated to the specific Receiver. If this list changes then the underlying [IS-04][IS-04] Receiver MUST update its version (in registered mode this MUST update the registered resource).
 
 If Receiver becomes unable to handle Flow, it is REQUIRED to take the following actions:
 - MUST set the [IS-05][IS-05] `/active`'s `master_enable` property to `false`.
 - MUST update the [IS-04][IS-04] `subscription`'s `active` property to `false` (regardless of unicast or multicast).
 
-## Sinks
+## Inputs
+
+The `/edid` endpoint allows a client to download the effective EDID if it exists. If the EDID information for the Input changes and it is associated with any Senders, then all of the Senders in question MUST update their versions (in registered mode this MUST update the registered resources).
```

Comment:
Why requiring Sender version change if EDID changes? If the EDID cause a change of Source or Flow the Sender will be updated, if the EDID changes without causing a change in Source or Flow then why update the version of the Sender? On the Input side the EDID changes through IS-11 so the controller knows about it.

Reply:
Right, it is. I assume, providing effective EDID binary to HDMI source is not immediate. Controller PUTs Media Profiles, the Sender responds and updates own version. Then the HDMI input sets HPD low, waits few hundreds of msecs, sets HPD high, the HDMI source reads the fresh effective EDID binary, the Input updates its effective EDID and updates own version and version of the Sender.

If effective EDID models not the last EDID read by HDMI source but the last EDID generated by Sender, we may treat Media Profiles change and effective EDID change as immediate. But I think bumping version of the Sender after effective EDID update is still useful for Inputs associated with multiple Senders and NMOS APIs clients other than the Controller.